### PR TITLE
qmake cleanup

### PR DESCRIPTION
**Pull request by ballessay**
Follow up on PR #41. Get the include and library paths from r2

### CODE REVIEW

File: src/lib_radare2.pri [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    -lr_cons \
+    -lr_io \
+    -lr_util \
+    -lr_flag \
+    -lr_asm \
+    -lr_debug \
+    -lr_hash \
+    -lr_bin \
+    -lr_lang \
+    -lr_io \
```
Comment (InusualZ) [inline src/lib_radare2.pri:43]: You have this library twice

File: src/Iaito.pro [hunk 2]
```diff
@@ -1,49 +1,46 @@
...
 CONFIG += c++11
 
+# Define the preprocessor macro to get the application version in our application.
+DEFINES += APP_VERSION=\\\"$$VERSION\\\"
 
-greaterThan(QT_MAJOR_VERSION, 4): QT += widgets
-
-TARGET = iaito
-TEMPLATE = app
+# why do we need to know the lib extension?
```
Comment (mrexodia) [inline src/Iaito.pro]: We don't, you can safely delete it.

File: src/lib_radare2.pri (hunk 1 above)
Comment (mrexodia) [reply src/lib_radare2.pri:43]: I would recommend sorting the libraries, makes it easier to spot:

```
LIBS += \
    -lr_anal \
    -lr_asm \
    -lr_bin \
    -lr_bp \
    -lr_config \
    -lr_cons \
    -lr_core \
    -lr_crypto \
    -lr_debug \
    -lr_egg \
    -lr_flag \
    -lr_fs \
    -lr_hash \
    -lr_io \
    -lr_lang \
    -lr_magic \
    -lr_parse \
    -lr_reg \
    -lr_search \
    -lr_socket \
    -lr_syscall \
    -lr_util
```

**Comment by mrexodia** [comment]
I tested on Window with MSVC and MinGW. Has my support!

File: src/lib_radare2.pri [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        } else {
+            error("sorry could not find radare2 lib")
+        }
+    } else {
+        error("r2 not found/in path")
+    }
+}
+
+# are really all of this libs needed?
+LIBS += \
```
Comment (trufae) [inline src/lib_radare2.pri]: should be the output of `pkg-config --libs r_core`

Comment (ballessay) [reply src/lib_radare2.pri]: It actually **is** the output of the command. Will delete the comment.

File: src/lib_radare2.pri (hunk 1 above)
Comment (ballessay) [reply src/lib_radare2.pri:43]: I just copied the the statement from Iaito.pro and it seems to be the output of pkg-config. Not sure if I should reorder them.

**Comment by trufae** [comment]
Better not to reorder because it is the linkage dependency order required to link with some linkers. It is possible to run pkgconfig instead of hardcoding this?

**Comment by ballessay** [comment]
@trufae good question. Should be doable, but here with user install pkg-config needs `PKG_CONFIG_PATH` set to generate the output. Note sure if this should be part of this PR

**Comment by ballessay** [comment]
@trufae A little problem with the pkg-config approach would be the Windows build. Does that toolchain provide pkg-config? I don't have a Windows installation here to test or verify. And in case it doesn't provide pkg-config the LIBS statement would be need there anyhow.

**Comment by mrexodia** [comment]
It doesn't.

**Comment by radare** [comment]
Oh right. Then we are fine just copypasting this from linux or mac's pkgconfig

**Comment by ballessay** [comment]
@probonopd A few questions which I haven't found after a short search. Why do we need to "sudo make install" + chown-ing it back to the $USER? And what is the reason to have the deskop file a second time in /usr/share/applications?

**Comment by probonopd** [comment]
@ballessay 

> Why do we need to "sudo make install" + chown-ing it back to the $USER?

That may in fact not be necessary, would be worth a try without it. Last time I tried, "make install" was not working entirely however and I had to do everything by hand.

> And what is the reason to have the deskop file a second time in /usr/share/applications?

We need one in the root of the directory because the AppImage specification needs it there, and it is customary to also have one in `usr/share/applications` which currently may not strictly be necessary but may enable future functionality, e.g., in the optional `appimaged` daemon.

**Pull request merged**